Task/111 implement backend for authorization of position names

tupito opened the pull request:
CodingTask: #111

Uudet apit:
* [x] GetPositionName: [GET] http://localhost:7071/positionnames/{id}
* [x] CreatePositionName: [POST] http://localhost:7071/positionnames
* [x] UpdatePositionName: [PUT] http://localhost:7071/positionnames/{id}

Muokatut apit:
* [x] GetPositionNames: [GET] http://localhost:7071/positionnames
* [x] CreatePosition: [POST] http://localhost:7071/positions
* [x] UpdatePosition: [PUT] http://localhost:7071/positions/{id}

EDIT: 3.4.2025 13:08:
Vain lisähuomiona, että `database/scripts/Inserts/PositionNames.sql` on vain demodataa, päivämäärät vedetty hatusta vain ja ainostaan testailua varten. (Tämä nyt vain lisähuomiona, kun virkanimike-excelistä on ollut nyt Teamsissa puhetta.)

tupito committed:
(#111) DB - Modify PositionNames and demodata to have ValidFrom and ValidUntil values

tupito committed:
(#111) Modify models for PositionNames

tupito committed:
(#111) Modify repository for PositionNames and changes to CreatePosition and UpdatePosition APIs

tupito committed:
(#111) Rename function namespace that PositionName model can be used for CreatePositionName API

tupito committed:
(#111) Add GetPositionName API

tupito committed:
(#111) Add CreatePositionName API

tupito committed:
(#111) Add UpdatePositionName API

tupito committed:
(#111) Fix CreatePosition and UpdatePosition to use PositionNameId instead of legacy PositionName object

tupito committed:
(#111) Comment out obsolete PositionName tests after logic refactor

JGrasscape reviewed (dismissed):
API:t toimii. Logiikka tosiaan muuttunut siten, että viran luonnissa ja päivittämisessä kuljetetaan mukana ainoastaan PositionNamen ID:tä, ei enää koko objektia. Tämä pitänee nyt sitten huomioida myös frontilla, esim. CreateVirkaModal.tsx ja modelissa Position.ts. Eli PositionNamet jatkossa samalla lailla kuin Costcentret.

Rikkoutuneet testit tosiaan kommentoitu nyt pois, pitää muistaa mainita Test reportiin tämä muutos, ellemme ehdi näitä vielä korjata.

Commit:
Disable mass change after mass change

* User need to reselect rows after mass change
* This is due to oversight that the position data changes, but selected
  rows contain same data. This caused the previous change to revert.
  This fixes that bug. Otherwise UX improvement so that the user knows
  what is wanted to be changed as mass change is not to be done often

Peskarhu commented:
Refaktorointi tehty frontille. Samalla päivitty logiikka parissa kohtaa. Yksi pieni bugi tuli vastaan ja tein nopeen fixin. Konflikteja pari ja ratkasin. Yleisesti pientä fixiä mitä tuli vastaan mm. localessa virhe.

Testattu massamuutokset, viran luominen, viran tietojen muokkaus ja viran tiedot näkymä OK.

Yleisesti kommenttina, kuin squashaat niin korjaa commit viestiä siistimmäksi eli esimerkkinä nyt jo mergetty Task/127 on:

"
```
Task/127 Implement backend for authorization of costcentres (https://github.com/virkarekisteri/virkarekisteri/pull/143)
* (https://github.com/virkarekisteri/virkarekisteri/issues/127) DB - Add Costcentres, delete OrgTree, modify positions

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Modify models for Costcentres

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Modify repositories for Costcentres

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Add and modify functions for Costcentres

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Modify and finish UpdateCostcentre repository to use CRUDChangeLogs

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Add CreateCostcentre api and modify repository

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Modify UpdatePosition API to use costcentres

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Modify CreatePosition API to use costcentres

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Modify CreatePositionsFromCsv API to use costcentres

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Fix various mentions of OrgTrees and such

* (https://github.com/virkarekisteri/virkarekisteri/issues/127) Fix tests to use costcentres

* Task/127 implement frontend changes

* Make adjustments without changing all of the orgtree to costcenter
* Works at this stage. Tested add, change/update, mass change and some
  filtering/sorting

* Bug fix

* Change the chip max height to not alter the input field height

* Renaming files and refactoring

* Renamed from orgtree to costcentre
* Renamed files
* Fixed some translations to use costcentre

* Fix lint issue I forgot

---------

Co-authored-by: Janne Ruohoniemi <janne.ruohoniemi@tuni.fi>
Co-authored-by: Lassi Lappalainen <lassi.lappalainen@tuni.fi>
```
"
olisi miellyttävämpi vain:

"
```
Task/127: Implement backend for costcentre authorization (#143)

* Added costcentres to the database, removed OrgTree, and updated positions accordingly.
* Refactored models, repositories, and functions to support costcentre operations.
* Integrated costcentres into CRUD workflows:
  - Updated UpdateCostcentre repository to use CRUDChangeLogs.
  - Updated/Create APIs for Position (CreatePosition, UpdatePosition, CreatePositionsFromCsv) to use costcentres.
* Fixed tests and resolved references previously pointing to OrgTree.
* Implemented corresponding frontend changes (Task/127) and made minor adjustments:
  - Corrected chip max-height so it does not affect the input field.
  - Renamed files and updated translations.
  - Fixed a linting issue.

---------

Co-authored-by: Janne Ruohoniemi <janne.ruohoniemi@tuni.fi>
Co-authored-by: Lassi Lappalainen <lassi.lappalainen@tuni.fi>
```
"

Pysyy siistinä logit ja hyvä palata jos pitää selvittää missä tapahtunut muutos versionhallinnassa.

Commit:
Refactoring frontend

* Rename positionname to use similar to costcenterid as positionnameid
* Refactoring logic to use new API
* Refactoring to not use object. Similar to the costcenter solution
* Fix translations
* Fix cases of old olgTreeId to use newer costcentreId
* Small fixes here and there

Commit:
Add isLoading

* Add isLoading to the names and costcenters queries

Commit:
Fix conflicts and small changes

* Fix issue with how costcenter active status was handled in UI after change
  to accept null values
* Fix insert script issue with ending dates. This does not fix the
  status bug, but makes the three examples work for testing purposes

tupito committed:
(#111) Trigger frozen pipeline checks
